src/base/project.lua

```diff
@@ -326,7 +326,9 @@
 		tree.trimroot(tr)
 		tree.sort(tr, sorter)
 
-		prj._.sourcetree = tr
+		if sorter then
```

Comment:
Should it be `not sorter` instead? (for cache consistency)
Or even better move it before the `table.sort`.
So `prj._.sourcetree` still have unsorted tree, and we sort the result afterward and `local tr = tree.new(prj.name)` creation should be done only when `prj._.sourcetree` is not `Nil`

Reply:
Oops, you're absolutely right, this should be `not sorter`.

But I also agree that your suggestion to cache unsorted tree is even better, done (and force pushed) now.

Thanks!

Reply:
Is this resolved?

Reply:
Yes, I think the latest version is optimal: it caches the unsorted tree and sorts it if necessary before returning it.